File: TesteTorniquete/src/Detector.java
import java.util.Scanner;

public class Detector {
    public EventoTorniquete detectar(){
        return processar();
    }

    public EventoTorniquete processar(){
        System.out.println("Evento:");
        Scanner scanner = new Scanner(System.in);
        String strEvento = scanner.nextLine();
        EventoTorniquete eventoTorniquete = null;
        switch (strEvento){
            case "CARTAO": eventoTorniquete = EventoTorniquete.CARTAO; break;
            case "PASSAGEM": eventoTorniquete = EventoTorniquete.PASSAGEM; break;
            case "REINICIAR": eventoTorniquete = EventoTorniquete.REINICIAR; break;
        }
        return eventoTorniquete;
    }
}

File: TesteTorniquetePython/torniquete/detector.py
from torniquete.evento import Evento


class Detector():
    def detectar(self):
        return self.processar()

    def processar(self):
        input_value = input("Evento: ")
        dicEventos = {"CARTAO": Evento.CARTAO,
                      "PASSAGEM": Evento.PASSAGEM,
                      "REINICIAR": Evento.REINICIAR}

        eventoTorniquete = dicEventos.get(input_value, '')
        return eventoTorniquete
